Zend/zend_execute_API.c

```diff
@@ -1010,7 +1010,11 @@ zend_result zend_call_function(zend_fcall_info *fci, zend_fcall_info_cache *fci_
 
 	if (UNEXPECTED(EG(exception))) {
 		if (UNEXPECTED(!EG(current_execute_data))) {
-			zend_throw_exception_internal(NULL);
+			if (Z_TYPE(EG(user_exception_handler)) != IS_UNDEF) {
```

Comment:
The fix looks simple, please consider https://3v4l.org/nXcjr usecase to target this PR to PHP 8.1 as bugfix, as currently there is no way, no workaround to handle exception in shutdown.

To be 100% BC break free, I propose:
for PHP 8.1 - 8.2: remove the exception handler before handling shutdown, but allow to register an exception handler in shutdown, so shutdown code executed after can be handled
for master: the current impl, ie. handler registered before shutdown can handle exception in shutdown (tested in `Zend/tests/gh10695_1.phpt`), LGTM

And 👍 for `Zend/tests/gh10695_4.phpt` and `Zend/tests/gh10695_5.phpt` tests.